metadata_writer_inventory_conn.c

```diff
@@ -604,7 +619,20 @@ static uint8_t md_inventory_usage_dump_db_json(struct md_writer_sqlite *mws, FIL
 {
     sqlite3_reset(mws->dump_usage);
 
-    return md_json_helpers_dump_write(mws->dump_usage, output);
+    json_object *jarray = json_object_new_array();
+
+    if (md_json_helpers_dump_write(mws->dump_usage, jarray) ||
+        json_object_array_length(jarray) == 0)
+    {
+        json_object_put(jarray);
+        return RETVAL_FAILURE;
+    }
+
+    const char *json_str = json_object_to_json_string_ext(jarray, JSON_C_TO_STRING_PLAIN);
```

Comment:
We should declare variables at the top of the function to be consistent with current code.